testkit/src/checkpoint_db.rs

```diff
@@ -106,6 +126,14 @@ impl<T: Database> CheckpointDbInner<T> {
     }
 
     fn merge(&mut self, patch: Patch) -> StorageResult<()> {
+        if self.checkpoint_set {
+            self.merge_with_journal_logging(patch)
+        } else {
+            self.db.merge(patch)
+        }
+    }
+
+    fn merge_with_journal_logging(&mut self, patch: Patch) -> StorageResult<()> {
```

Comment:
I think `merge_with_logging` is better. `Journal` and `log` is the same thing, in fact.